server/src/user/subscriptions/subscriptions.service.ts

```diff
@@ -39,10 +39,23 @@ export class SubscriptionsService {
     @InjectQueue('subscriptions')
     private subscriptionsQueue: Queue<SubscriptionsQueueParams>,
     private notificationsService: NotificationsService,
-    private readonly logger: Logger
-  ) {}
+    private readonly logger: Logger,
+    private schedulerRegistry: SchedulerRegistry
+  ) { }
+
+  async initializeSubscriptionTask() {
+    const timeInMilliseconds = this.getSubscriptionIntervalTime() * 1000 * 60;
+    const interval = setInterval(this.collectSubscriptionsJob, timeInMilliseconds);
```

Comment:
```suggestion
    const interval = setInterval(this.collectSubscriptionsJob.bind(this), timeInMilliseconds);
```

I found a fix for when running in dev mode with `pnpm serve`

Reply:
oh that's an interesting fix. I wondered how it works without this fix anyway, but I assumed that `this.schedulerRegistry.addInterval` handles cases where this is necessary.
Can you test with `pnpm build && pnpm run start`  without the fix?
I would like to know if somehow there is a race condition or something or if it really is the difference between `serve` and `build`. Because if it is, I should really get the `serve` running on my end.

Reply:
The error happens aswell for me with pnpm build && pnpm run start. 
If I set the time to 1 min, the first run succeeds (I presume because it is called directly) and the subsequent runs fail.

What is the problem you are running into when running serve? I may be able to help you.

Reply:
I get the following error when executing `pnpm run serve:server`:
![image](https://github.com/ViewTube/viewtube/assets/12413072/0ad1425a-2371-49f3-b8a5-d0345be5642b)

I also get the same error when running `pnpm run serve` from within the `server` directory. So I think it's not a path issue.

Reply:
I actually got the same error after the first run. So it doesn't depend on the setup and is just that `.bind(this)` is missing.

Reply:
added the `bind(this)`

Reply:
> I also get the same error when running `pnpm run serve` from within the `server` directory. So I think it's not a path issue.

Ah it's the one I was expecting :sweat_smile:. I don't know what causes it but I have a fix, I'll merge it into development after this PR.